packages/core/src/index.ts

```diff
@@ -63,18 +63,18 @@ export function plugins(ps: Plugin[]): Plugin {
 export class Compiler {
     private fallbackBindingsPluginWarningDisplayed = false;
 
-    async minify(src: string, opts?: JsMinifyOptions, extras?: NapiMinifyExtra): Promise<Output> {
+    async minify(src: string | Buffer, opts?: JsMinifyOptions, extras?: NapiMinifyExtra): Promise<Output> {
         if (bindings) {
-            return bindings.minify(toBuffer(src), toBuffer(opts ?? {}), extras ?? {});
+            return bindings.minify(Buffer.from(typeof src === 'object' ? JSON.stringify(src) : src), toBuffer(opts ?? {}), typeof src === 'object', extras ?? {});
```

Comment:
@kdy1 
You widened the allowed type of `src` to include `Buffer`, in which case `typeof src === 'object'`, which means if a caller passes a `Buffer` it will get stringified, which is going to make a mess. If you want to include `Buffer` in the allowed input types you'll want to handle it explicitly.

Reply:
Ah you are right
Thank you!